src/Processors/Processor.cs

```diff
@@ -0,0 +1,62 @@
+﻿namespace Processors;
+
+public abstract class Processor : BaseThread
+{
+    private bool _isPaused = false;
+    private bool _onPauseOnce = false;
+    private bool _onResumeOnce = false;
+    private bool _stopRequested = false;
+    private int _interval = 500;
+    public ProcessorStatus status = ProcessorStatus.None;

```

Comment:
if you have this, why do you have `_isPaused`, `_onPauseOnce`, `_onResumeOnce`, `_stopRequested`?

Reply:
I understand your point.

- I use `_isPaused` and `_stopRequested` to decide when I have to change the status of the thread (as a transition).
- `ProcessStatus` gives me the actual status of the thread.
- Variables `_onPauseOnce` and `_onResumeOnce` helps me to invoke `OnPause()` and `OnResume()` once.

Something I think that causes confusion is the name `_isPaused` which I could change it to something like `_pauseRequested`.

Reply:
I reworked this class and I'm not using these variables. Please check the updated version.